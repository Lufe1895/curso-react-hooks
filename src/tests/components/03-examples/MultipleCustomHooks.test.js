import { shallow } from "enzyme"
import { MultipleCustomHooks } from "../../../components/03-examples/MultipleCustomHooks"
import { useFetch } from "../../../hooks/useFetch";

jest.mock("../../../hooks/useFetch");

describe('Pruebas sobre multiple custom hooks', () => {
    test('Debe mostrarse correctamente', () => {
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper).toMatchSnapshot();
    });
   
    test('Debe mostrar la informaciÃ³n', () => {
        useFetch.mockReturnValue({
            data: [{
                author: 'Luis',
                quote: 'Holas'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper.find('.alert').exists()).toBe(false);
        expect(wrapper.find('.mb-0').text().trim()).toBe('Holas');
        expect(wrapper.find('footer').text().trim()).toBe('Luis');
    });
    
});
