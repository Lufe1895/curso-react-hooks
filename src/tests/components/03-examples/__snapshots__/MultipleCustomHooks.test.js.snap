// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre multiple custom hooks Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading
  </div>
</div>
`;
